### ISSUE TITLE
feat: add custom namespace to Stripe SDK using PHPScoper

### ISSUE THREAD

**Issue by mehul0810** [impress-org/give#4186]
# Feature Request

## User Story
As a user, I want to ensure that the Stripe SDK via composer used in Give doesn't conflict with the other plugins using the same Stripe SDK with different version so that it will provide a better UX for the users.

## Possible Solution
Using PHP Scoper can help to add custom namespace to the Stripe SDK loaded via composer

## Research
In my initial research, I noticed that the adding custom namespace using PHP Scoper to the Stripe SDK doesn't replace some of the use statements implemented within it which throws errors when we call some of the Stripe API functions.

So, I've tried to read the documentation on the GitHub repo of PHP Scoper and done some research on how to tackle this scenario and came to know about **Patchers**. Patchers is a great configuration provided in PHP Scoper which on initial research didn't look so promising. But, when I have proceed into more details it proved to be very useful when some of the functions or statements within the Stripe SDK doesn't work.

Now, using Patchers, I'm able to replace those left out statements using `preg_replace` and `str_replace` which worked like charm. Also, I tried couple of Stripe API calls using the Give name-spaced Stripe SDK and it worked fine.

So, I'll now proceed with the steps mentioned in the acceptance criteria to see whether this works with actual Give and Stripe setup or not. I'll add further research and insights into this issue.

## Related
- https://github.com/impress-org/give/issues/4180
- https://github.com/humbug/php-scoper

## Acceptance Criteria
- [x] Generate Stripe SDK with custom namespace
- [x] Replace the strings which is throwing error using Patchers
- [x] Automate the process of generating the custom Give name-spaced Stripe SDK using webpack
- [ ] Testing

**Comment by mehul0810** [comment]
## Slack Call Summary
Participants: @ravinderk @mehul0810 
Topic: Discussion about better folder structure with PHP Scoper for Stripe SDK
Result: I and Ravinder had a discussion about PHP Scoper with Stripe SDK and how to manage it with better folder structure after doing search replace via PHP Scoper. So, I moved running php scoper from `package.json` to `composer.json` and used `webpack` to clean and copy the custom name-spaced Stripe SDK to `vendor` directory using `npm run dev` or `npm run production`. Also, I'll do more research on Clean WebPack Plugin to ensure that we delete the output directory generated via PHP Scoper.

**Comment by mehul0810** [comment]
As discussed over Slack call, I'll be working on the 2 points to ensure better backward compatibility:
1. Use the original Stripe SDK with direct call for backward compatibility and Use the prefixed version of Stripe SDK with composer
2. Use original Stripe SDK and prefixed Stripe SDK in parallel using composer to ensure that backward compatibility is retained.

I'll work on these points and will revert back with detailed findings.

**Comment by mehul0810** [comment]
@DevinWalker @mikejhale I've done further research on this issue and it seems that above mentioned both the scenarios are not possible for adding backward compatibility.

However, I've got a work-around to resolve the issue with backward compatibility. But, it will not convert the Stripe SDK with custom namespace.

We can resolve the issue just by triggering the Stripe SDK manually, instead of loading via composer. I've tested it with EDD Stripe add-on as well as WP Full Stripe WordPress plugin which has past compatibility issues with Give Stripe add-on. 

Let's discuss about this further in today's call to finalize on this issue.
